File: app/src/main/java/com/route/recyclerview_online/SettingsAdapter.java
package com.route.recyclerview_online;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class SettingsAdapter extends RecyclerView.Adapter<SettingsAdapter.SettingsViewHolder> {
    public SettingsAdapter( ArrayList<SettingsItem> items) {
        this.items = items;
    }

    ArrayList<SettingsItem> items;

    @NonNull
    @Override
    public SettingsViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        View view = LayoutInflater.from(parent.getContext()).
                inflate(R.layout.item_settings, parent, false);
        return new SettingsViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull SettingsViewHolder holder, int position) {
        SettingsItem item = items.get(position); // 0
        holder.titleTextView.setText(item.title);
        holder.subtitleTextView.setText(item.subtitle);
        holder.settingsIcon.setImageResource(item.imageId);
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    class SettingsViewHolder extends RecyclerView.ViewHolder{
        TextView titleTextView,subtitleTextView;
        ImageView settingsIcon;
        public SettingsViewHolder(@NonNull View itemView) {
            super(itemView);
            titleTextView = itemView.findViewById(R.id.item_settings_title);
            subtitleTextView = itemView.findViewById(R.id.item_settings_subtitle);
            settingsIcon = itemView.findViewById(R.id.item_settings_icon);
        }
    }
}

File: app/src/main/java/com/route/recyclerview_online/kotln/SettingsAdapter.kt
package com.route.recyclerview_online.kotln

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.route.recyclerview_online.R
import java.util.*

class SettingsAdapter(var items: ArrayList<SettingsItem>): RecyclerView.Adapter<SettingsAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
      val view : View = LayoutInflater.from(parent.context).
      inflate(R.layout.item_settings, parent, false)
       return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = items.get(position)
        holder.title.text = item.title
        holder.subTitle.text = item.subtitle
        holder.settingsIcon.setImageResource(item.imageId)
        holder.settingsIcon.setOnClickListener{view->
            ///todo: write your logic
           onIconClick?.onItemClick(item, position)

        }
        holder.settingsIcon.setOnClickListener(object :View.OnClickListener{
            override fun onClick(p0: View?) {
             /// todo: write your logic here
            }
        })
    }
    var onIconClick: OnItemClickListener? = null

    interface OnItemClickListener{
        fun onItemClick(item: SettingsItem, position: Int)
    }
    override fun getItemCount(): Int {
       return items.size
    }
    class ViewHolder(var itemView: View) : RecyclerView.ViewHolder(itemView){
        val settingsIcon: ImageView = itemView.findViewById(R.id.item_settings_icon)
        val title: TextView= itemView.findViewById(R.id.item_settings_title)
        val subTitle: TextView = itemView.findViewById(R.id.item_settings_subtitle)
    }
}